### DISCUSSION TITLE
[Detector Support]: Coral TPU & GPU

### DISCUSSION THREAD

**Discussion by MrCaspan** [blakeblackshear/frigate#15039] [Detector Support]
### Describe the problem you are having

I have a USB Coral TPU and a Nvidia GTX 1080 with -tensor image loaded .  Is it worth having both available for decoding or should I only use the Coral TPU. I have 6 cameras 3x 4K, 2x 2K 1x 1K Just not sure if its wroth having both running or not

### Version

0.14.1

### Frigate config file

```yaml
N/A
```

### docker-compose file or Docker CLI command

```yaml
N/A
```

### Relevant Frigate log output

```shell
N/A
```

### Install method

Docker Compose

### Object Detector

Coral

### Screenshots of the Frigate UI's System metrics pages

N/A

**Comment by hawkeye217** [comment]
What inference times are you seeing?

**Comment by MrCaspan** [comment]
I was seeing ~10ms with GPU TPU and ~10ms with Coral TPU

But there is 10% gpu memory used but 0 GPU power is used

Not sure if this helps?

![image](https://github.com/user-attachments/assets/defb02b5-4c5c-4f38-8e34-ea11b9ea5f6b)

**Reply by hawkeye217** [reply]
I'd say you'd be fine with either one. The Coral will definitely be more power-efficient. For some features in Frigate 0.15 and later, your GPU may be useful.

**Reply by MrCaspan** [reply]
Is there any ham having both available. I am assuming not?

**Reply by hawkeye217** [reply]
No harm, Frigate will only use one, though.

**Reply by MrCaspan** [reply]
Wow so I enabeled detect and motion support with both the GPU and Coral TPU and bot still increase the CPU by an average of13%. Normal CPU for the box is 19% and now is 30-33%.. I thought that the TPU will handle most of this stuff?Is it really this CPU intensive still?

**Reply by NickM-27** [reply]
You seem to be misunderstanding. The coral can do one thing and one thing only and that is run inferences from machine learning models. The coral does not do any decoding or motion detection

**Reply by NickM-27** [reply]
Frigate also does not support running multiple object detectors at the same time so not sure how you're using a tpu and gpu at the same time.

**Reply by MrCaspan** [reply]
Sorry Nick I meant I did a tested with both of them, IE not at the same time but one at a time to see if one performed better. I am still supprised on the CPU raise of 13%. on either of them. So I am guessing the motion detection is causing the gain in CPU. Let me test with each. with motion off

NM I beleive detection requires motion to be on.. Ill double check but the CPU did settle down over the last hour

**Reply by MrCaspan** [reply]
I think I know why CPU is high now. I tested the stream using the API URL and this is what I see.... i think it's a little too sensative

![image](https://github.com/user-attachments/assets/f85ad224-1140-4ceb-bffc-0d64d73bd646)

**Reply by NickM-27** [reply]
Yeah seems like it
